DataFormats/Portable/interface/PortableDeviceCollection.h

```diff
@@ -79,10 +83,29 @@ class PortableDeviceCollection {
     alpaka::memset(std::forward<TQueue>(queue), *buffer_, 0x00);
   }
 
+  // Copy column by column heterogeneously for device to host data transfer.
+  template <typename TDescriptor, typename TQueue>
```

Comment:
What restrictions are there on the `TDescriptor` type ?
Can it be only `Layout::ConstDescriptor` or `Layout::Descriptor` ?
Or can it really be arbitrary ?

Reply:
If I understand the code correctly, the `Descriptor` can be constructed directly from a `View`.
If that is the case, could `deepCopy` be written to take a `View` or `ConstView` as argument, and build the descriptor form it internally, before calling `_deepCopy` ?